parser/ast/ddl.go

```diff
@@ -1689,7 +1690,13 @@ type DropIndexStmt struct {
 func (n *DropIndexStmt) Restore(ctx *format.RestoreCtx) error {
 	ctx.WriteKeyWord("DROP INDEX ")
 	if n.IfExists {
-		ctx.WriteKeyWord("IF EXISTS ")
+		if ctx.Flags.HasTiDBSpecialCommentFlag() {
+			ctx.WriteWithSpecialComments("", func() {
```

Comment:
the `if` check is already handled in `WriteWithSpecialComments`. no need to check it again here.

Reply:
https://github.com/pingcap/tidb/pull/30173/commits/eb37727dce871522a23f336d330bad3bbe3e77e1 the output has an extra space :joy: